x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportGetDataFrameTransformsStatsAction.java

```diff
@@ -224,6 +242,37 @@ private void collectStatsForTransformsWithoutTasks(Request request,
             searchStatsListener, client::search);
     }
 
+    private void getCheckpointingStateAndStatsWithoutTask(ClusterState state, DataFrameIndexerTransformStats stats,
+            ActionListener<DataFrameTransformStateAndStats> listener) {
+        PersistentTasksCustomMetaData pTasksMeta = state.getMetaData().custom(PersistentTasksCustomMetaData.TYPE);
+
+        if (pTasksMeta == null) {
+            listener.onResponse(DataFrameTransformStateAndStats.initialStateAndStats(stats.getTransformId()));
+        }
+
+        Collection<PersistentTasksCustomMetaData.PersistentTask<?>> existingTask = pTasksMeta.findTasks(DataFrameTransform.NAME,
+                t -> t.getId().equals(stats.getTransformId()));
+
+        if (existingTask.isEmpty()) {
+            listener.onResponse(DataFrameTransformStateAndStats.initialStateAndStats(stats.getTransformId()));
+        }
```

Comment:
I do not think this `if` branch is correct. It is plausible that no persistent tasks are in the cluster state, but there still be `DataFrameIndexerTransformStats` stored in a doc. If this is the case, this branch will bypass that situation and not return the stats.